elegy/model/generalized_module/__init__.py

```diff
@@ -0,0 +1,2 @@
+from . import elegy_module, linen_module
```

Comment:
Is the dependency on Flax necessary? Can we make it optional?

Reply:
It will be optional. I will add some checks for the dependent modules (flax, haiku, ect) to do conditional imports of the Generalized modules here.